### PR TITLE
a small commit to close the plots for better memory usage

### PR DESCRIPTION
**Pull request by marjanfamili**

### CODE REVIEW

**Comment by marjanfamili** [comment]
#77

**Review by crangelsmith — APPROVED** [review]

File: avae/vis.py [hunk 1]
```diff
@@ -230,10 +274,33 @@ def loss_plot(epochs, train_loss, val_loss=None, p=None):
...
+    plt.legend()
+
     plt.tight_layout()
     if not os.path.exists("plots"):
         os.mkdir("plots")
     plt.savefig("plots/loss.png")
+    plt.close()
+
+    # only training loss
+    for i, loss in enumerate(train_loss):
```
Comment (jolaem) [inline avae/vis.py:292]: Is this to have a separate loss for training data for clarity of display?

Comment (marjanfamili) [reply avae/vis.py:292]: Yes, this is something I was going to discuss with you. Do you think in a paper we only need to show the training loss or also validation loss ? It's just that the loss plot although excellent for us, do you think looks really busy for publication ?

Comment (jolaem) [reply avae/vis.py:292]: I agree that it looks busy.

Having said that - showing validation plot is important because it can show overfitting.

But I do agree it looks a bit busy so for a publication maybe we could show the full plot side by side with a TOTAL loss plot only? That makes it less busy and a bit more clear while still illustrating the overfitting situation. What do you think?

**Comment by jolaem** [comment]
Also just heads up that affinity matrix is now looking a bit busy for a large number of classes. Maybe increasing figure size would help?
Affinity_Matrix

File: avae/vis.py (hunk 1 above)
Comment (marjanfamili) [reply avae/vis.py:292]: I think then figure size should depend on the number of classes ?

**Comment by marjanfamili** [comment]
#74 #76 #77 #75

**Review by jolaem — APPROVED** [review]
